manifests/params.pp

```diff
@@ -64,5 +64,7 @@
   $pulpcore_postgresql_ssl_cert    = '/etc/pki/katello/certs/pulpcore-database.crt'
   $pulpcore_postgresql_ssl_key     = '/etc/pki/katello/private/pulpcore-database.key'
   $pulpcore_postgresql_ssl_root_ca = '/etc/pki/tls/certs/ca-bundle.crt'
+  $pulpcore_allowed_import_path    = ['/var/lib/pulp/sync_imports', '/var/lib/pulp/imports']
+  $pulpcore_allowed_export_path    = ['/var/lib/pulp/exports']
```

Comment:
```suggestion
  $pulpcore_allowed_import_path    = ['/var/lib/pulp/sync_imports']
  $pulpcore_allowed_export_path    = []
```

Let's set defaults here that most appropriate for the Content Proxy scenario (i.e. Capsule). This also makes sense to do, since the migration you have opened in https://github.com/theforeman/foreman-installer/pull/615 is for the Katello scenario only. So FPC scenario will have no migration and therefore just get the defaults for this module which we define here. Then for the Katello scenario we'll override the defaults with the installer migration and give the values that Katello needs instead.

I think this makes the most logical sense, and as an added benefit, it's consistent with what you are already doing at the installer level. :+1:

Reply:
A different approach would be to hardcode it in `init.pp`, depending on whether it's a mirror or not. That would avoid the migrations in the installer itself. While it provides less flexibility for the user, it does make it easier to consume.

Reply:
What is being suggested here is more so to drop the parameters from being user defined and instead set them just prior to this based on the `pulpcore_mirror` condition, something like:

```
  if $pulpcore_mirror {
    $pulpcore_allowed_import_path    = ['/var/lib/pulp/sync_imports']
    $pulpcore_allowed_export_path    = []
  } else {
    $pulpcore_allowed_import_path    = ['/var/lib/pulp/sync_imports', '/var/lib/pulp/imports']
    $pulpcore_allowed_export_path    = ['/var/lib/pulp/exports']
  }
```

So if its a mirror, effectively set them to the defaults, otherwise if main Pulp set them to these needed import/export paths. And do this in `init.pp` before the pulpcore class is declared and not in params.pp.